packages/react-core/src/components/Select/Select.tsx

```diff
@@ -400,6 +400,9 @@ export class Select extends React.Component<SelectProps & OUIAProps, SelectState
                 }
               })
             : childrenArray;
+        if (typeaheadInputValue.toString() !== '') {
```

Comment:
On line 386 there is a comment about not filtering out dividers. Could that be removed and the logic be modified there to filter out dividers?

Reply:
I think it would be very convoluted, inside the `filter` callback I cannot tell which of preceding elements have been filtered out.

Reply:
I dont think it's more convoluted? 
If you change the comment on line 386 to say `// Dividers don't have value and SHOULD be filtered`
and then change the return statement on line 388 to `return false`

then likely you wont need another whole function call after having already iterated over every child.

Reply:
Well, yes but I want to keep the dividers in, just not all of them.
I.e. instead of the filtered result looking like `abc | | abd | | | abx |` (current implementation) 
I want it to look like `abc | abd | abx`.

Reply:
@michpetrov I see - yeah I did find it a little confusing given the comments a few lines up. Maybe you could update the comment to say something like 'filtering out dividers is handled after determining which values are filtered out. And then add a comment before your new `if` statement to say something like 'filter out unnecessary dividers'